map/src/drawer/components/tracks/TracksMenu.jsx

```diff
@@ -0,0 +1,105 @@
+import React, {useState, useContext, useEffect} from 'react';
+import {Typography, ListItemText, Collapse, MenuItem, ListItemIcon, LinearProgress, Button} from "@mui/material";
+import {DirectionsWalk, ExpandLess, ExpandMore} from '@mui/icons-material';
+import AppContext from "../../../context/AppContext"
+import TrackGroup from "./TrackGroup";
+import VisibleTrackGroup from "./VisibleTrackGroup";
+import LocalTrackGroup from "./LocalTrackGroup";
+
+
+export default function TracksMenu() {
+
+    const ctx = useContext(AppContext);
+
+    const [gpxOpen, setGpxOpen] = useState(false);
+    const [gpxFiles, setGpxFiles] = useState([]);
+    const [tracksGroupsOpen, setTracksGroupsOpen] = useState(false);
+    const [tracksGroups, setTracksGroups] = useState([]);
+    const [visibleTracks, setVisibleTracks] = useState({created: [], cloud: []});
+
+    //get gpx files and create groups
+    useEffect(() => {
+        let files = (!ctx.listFiles || !ctx.listFiles.uniqueFiles ? [] :
+            ctx.listFiles.uniqueFiles).filter((item) => {
+            return (item.type === 'gpx' || item.type === 'GPX')
+                && (item.name.slice(-4) === '.gpx' || item.name.slice(-4) === '.GPX');
+        });
+        files.forEach(f => {
+            f.folder = f.name.includes('/') ? f.name.split('/')[0] : 'Tracks';
+        });
+        setGpxFiles(files);
+
+        files.forEach(f => {
+            let group = tracksGroups.find(g => {
+                return g.name === f.folder;
+            })
+            if (group) {
+                group.files.push(f);
+            } else {
+                tracksGroups.push({name: f.folder, files: [f]});
+            }
+        });
+
+        if (tracksGroups.length > 0) {
+            let defGroup = tracksGroups.find(g => {
+                return g.name === 'Tracks';
+            })
+            if (defGroup) {
+                tracksGroups.splice(tracksGroups.indexOf(defGroup), 1);
+                tracksGroups.unshift(defGroup);
+            }
+        }
+
+        setTracksGroups(tracksGroups);
+
+    }, [ctx.listFiles, ctx.setListFiles]);
+
+
+    useEffect(() => {
+        if (ctx.gpxFiles) {
+            visibleTracks.cloud = [];
```

Comment:
setVisibleTracks looks strange it should be no difference cloud or not cloud is visible.

Reply:
Cloud and localClientTrack are very different objects with very different fields. We can't work with them same way. For cloud tracks and localServerTrack will be created selectedGpxFile for work with them. I dont change it.